### DISCUSSION TITLE
Get actual donation ID (Wordpress Post ID) from Sequential ID

### DISCUSSION THREAD

**Discussion by 8verstudio-Chua** [impress-org/givewp#7303] [Developer Q&A]
Hi, 

I'm developing a custom plugin that includes a function to retrieve donation details. However, the ID I'm getting is the sequential ID, not the default donation ID that follows the WordPress post ID order. So when I attempt to retrieve the donation details using the code below:
`` give()->donations->getById($donationID);``
I'm unable to retrieve the details.

Upon further investigation, I realized that the $donationID I'm receiving is 1, which corresponds to the GiveWP sequential ID, not the actual donation ID.

My question is, if I can only obtain this sequential ID, how can I retrieve the actual donation ID to get the donation details?
<img width="630" alt="Screenshot 2024-03-13 at 4 13 28 PM" src="https://github.com/impress-org/givewp/assets/75621599/1f1aaa5c-9253-4a2e-8b18-def1136d0c1b">

**Comment by jonwaldstein** [comment]
Hey @8verstudio-Chua 

Thanks for reaching out!

You can try this:

`give()->sequential_donation_db->get_column_by('payment_id', 'id', $sequentialId)`

Let me know if that works for you.

I would also love to learn more about that custom plugin you are building, if you want to chat more about your integration please email me at `jon@givewp.com` and we can setup a time to chat 😄
